nntrainer/src/flatten_layer.cpp

```diff
@@ -23,13 +23,32 @@ namespace nntrainer {
 
 int FlattenLayer::initialize(bool last) {
   int status = ML_ERROR_NONE;
-  // NYI
+  if (input_dim.getDataLen() == 1) {
+    ml_logw("Warning: the length of previous layer dimension is one");
+  }
+  if (input_dim.batch() <= 0 || input_dim.height() <= 0 ||
+      input_dim.width() <= 0 || input_dim.channel() <= 0) {
+    ml_loge("Error: Dimension must be greater than 0");
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+
+  this->last_layer = last;
+
+  output_dim.batch(input_dim.batch());
+  output_dim.channel(1);
+  output_dim.height(1);
+  output_dim.width(input_dim.getFeatureLen());
+
+  hidden = Tensor(output_dim);
+
   return status;
 }
 
 Tensor FlattenLayer::forwarding(Tensor in, int &status) {
-  // NYI
-  return in;
+  memcpy(hidden.getData(), in.getData(),
+         in.getDim().getDataLen() * sizeof(float));
```

Comment:
How about adding new method `getDataSize()` to `Tensor`?

Reply:
right. I will do it another PR. Thanks.